src/routes/index.ts

```diff
@@ -0,0 +1,30 @@
+/* @jsx createElement */
+import { NotFound } from "..";
+import { createElement, render } from "../render";
+import { routes } from "./routes";
+
+export const navigateTo = (url: string) => {
+  const isSamePath = location.pathname === url;
+  if (isSamePath) {
+    history.replaceState(null, "", url);
+  } else {
+    history.pushState(null, "", url);
+  }
+  router();
+};
+
+export const router = async () => {
+  const matchedRoutes = routes.map((route) => {
+    return {
+      route,
+      isMatch: location.pathname === route.path,
+    };
+  });
+
+  const match = matchedRoutes.find((matchedRoute) => matchedRoute.isMatch);
```

Comment:
```suggestion
  const matchedRoute = matchedRoutes.find((matchedRoute) => matchedRoute.isMatch);
```
match는 행위를 의미하는 동사이므로 `matchedRoute`처럼 연산의 결과가 반환한 값을 표현하는 이름이 더 정확할 것 같습니다